pkg/kubelet/kubelet_node_status.go

```diff
@@ -1028,9 +1040,9 @@ func (kl *Kubelet) validateNodeIP() error {
 		case *net.IPAddr:
 			ip = v.IP
 		}
-		if ip != nil && ip.Equal(kl.nodeIP) {
+		if nodeIP != nil && ip.Equal(nodeIP) {
```

Comment:
Typo? Should this be ip != nil?

Reply:
It should be. I will update and resubmit the PR. Thanks.